src/diracx/db/jobs/schema.py

```diff
@@ -198,3 +210,99 @@ class LoggingInfo(JobLoggingDBBase):
     StatusTimeOrder = Column(Numeric(precision=12, scale=3), default=0)
     StatusSource = Column(String(32), default="Unknown")
     __table_args__ = (PrimaryKeyConstraint("JobID", "SeqNum"),)
+
+
+class TaskQueues(TaskQueueDBBase):
+    __tablename__ = "tq_TaskQueues"
+    TQId = Column(Integer, primary_key=True)
+    Owner = Column(String(255), nullable=False)
+    OwnerDN = Column(String(255))
+    OwnerGroup = Column(String(32), nullable=False)
+    VO = Column(String(32), nullable=False)
+    CPUTime = Column(BigInteger, nullable=False)
+    Priority = Column(Float, nullable=False)
+    Enabled = Column(Boolean, nullable=False, default=0)
+    __table_args__ = (Index("TQOwner", "Owner", "OwnerGroup", "CPUTime"),)
+
+
+class JobsQueue(TaskQueueDBBase):
+    __tablename__ = "tq_Jobs"
+    TQId = Column(
+        Integer, ForeignKey("tq_TaskQueues.TQId", ondelete="CASCADE"), primary_key=True
+    )
+    JobId = Column(Integer, primary_key=True)
+    Priority = Column(Integer, nullable=False)
+    RealPriority = Column(Float, nullable=False)
+    __table_args__ = (Index("TaskIndex", "TQId"),)
+
+
+class SitesQueue(TaskQueueDBBase):
+    __tablename__ = "tq_TQToSites"
+    TQId = Column(
+        Integer, ForeignKey("tq_TaskQueues.TQId", ondelete="CASCADE"), primary_key=True
+    )
+    Value = Column(String(64), primary_key=True)
+    __table_args__ = (
+        Index("SitesTaskIndex", "TQId"),
+        Index("SitesIndex", "Value"),
+    )
+
+
+class GridCEsQueue(TaskQueueDBBase):
+    __tablename__ = "tq_TQToGridCEs"
+    TQId = Column(
+        Integer, ForeignKey("tq_TaskQueues.TQId", ondelete="CASCADE"), primary_key=True
+    )
+    Value = Column(String(64), primary_key=True)
+    __table_args__ = (
+        Index("GridCEsTaskIndex", "TQId"),
+        Index("GridCEsValueIndex", "Value"),
+    )
+
+
+class BannedSitesQueue(TaskQueueDBBase):
```

Comment:
Would you check if this is still used anywhere?

Reply:
This is still used by the JobScheduling optimizer (__sendToTQ method that create/fills the JobRequirements section of the JDL, which is then interpreted by the insertIntoTQ method of the JobState class which calls the TQDB insertJob method). This can be the case if a user specifies a list of banned sites and the job doesn't require any site in particular.

However, I would argue that having a BannedSitesQueue table is not very appealing. 

One solution would be to get the list of all sites, remove from this list the banned sites and insert the remaining sites in the `tq_TQToSites` table.

I think we should also ask the community if banning sites is a real use case, because this feature brings quite a lot of complexity.

Reply:
This feature is there because, most probably, of some (LHCb) user requirement of long time ago. This reminds me that we have often argued that it did not make much sense and that could even create issues. I would be tempted to remove it also.

Reply:
As long as we use the ON DELETE CASCADE constraint in DIRAC to remove the entries from this table, then I think we can remove this table from diracx.